source/CMakeLists.txt

```diff
@@ -426,7 +426,3 @@ if(ENABLE_SPIRV_TOOLS_INSTALL)
   install(FILES ${CMAKE_BINARY_DIR}/${SPIRV_TOOLS}Config.cmake DESTINATION ${PACKAGE_DIR})
 endif(ENABLE_SPIRV_TOOLS_INSTALL)
 
```

Comment:
Would it be possible to keep the `/MP4` if this is the top level project?

Reply:
I mean you could, but it would still break people building with ninja and msvc.

Also sometimes the `-m` option to msbuild is better than `/MP`

(recalling from a conversation I had with a compiler engineer long ago)
back in the day msvc was quite slow to startup and initialize, so slow in fact that it could actually matter even after doing the work of compiling. Thus /MP was added. Nowadays msvc is much faster to start up.

You can actually customize /MP after the fact when running msbuild using /p:CLMPCount=4 or similar and this is what the IDE setting customizes.

Maybe the solution is 
`set_property(TARGET <target> PROPERTY VS_GLOBAL_UseMultiToolTask true)`
you can also
`set_property(TARGET <target> PROPERTY VS_GLOBAL_EnforceProcessCountAcrossBuilds true)`
which will cause msbuild and cl to be aware of eachother, otherwise you will use 2**coreCount cores.

these also don't have to be conditional on msvc or the visual studio generator, you can just set these properties on your targets and be done with it. I'm not sure if there's a global way to set them.

Reply:
I added this to improve the parallelism in the Vulkan CTS build (https://github.com/KhronosGroup/VK-GL-CTS). As far as I can tell, reverting this will just slow things back down. At least with the versions of Visual Studio I've used, you get N projects built in parallel, and each project uses 1(ish) core. If you specify /MP, then each project will use N cores, and you get N^2 processes and the system may become unusable. I chose /MP4 because it was "good enough" and because a worst case of 4*N cores probably isn't going to be problematic.

EnforceProcessCountAcrossBuilds looks to be pretty new, and if this allows using /MP everywhere then I'd suggest making a Vulkan CTS PR to enable /MP and EnforceProcessCountAcrossBuilds, conditional on MSVC version, and then to make this change also conditional on MSVC version.

Reply:
you could also use 
`set_property(TARGET <target> PROPERTY VS_GLOBAL_CLMPCores 4)`

instead of setting the /MP option, this will make it so that the option is only used for the msbuild based generators, not the generators that already know how to do multiple rules at once. This is an older option.

Reply:
Is this option documented somewhere? I googled but can't find anything about it.